ecl/hthor/hthor.cpp

```diff
@@ -8289,22 +8254,40 @@ bool CHThorDiskReadBaseActivity::openNext()
                 filelist.append('\n').append(file);
                 try
                 {
-                    inputfile.setown(createIFile(rfilename));   
-                    if(compressed)
+                    inputfile.setown(createIFile(rfilename));
+                    if (rfilename.isLocal() && (!canSerialize || !readRemote(rfilename)))
```

Comment:
Is this test correct?
Currently xml/csv needs to be read local - even if it is not a local filename
Similarly if you cannot serialize then it needs to be local.

Possibly:
`    if (!isBinary() || !canSerialize || (isLocal && !readRemote())
`

Reply:
Agreed, also going to change readRemote() to forceRemote() for clarity.